[MRG] Use 'latin-1' decoding for GDF header fields

cbrnr opened the pull request:
Otherwise loading a GDF file results in `UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf6 in position 5: invalid start byte` or similar errors.

agramfort commented:
is Latin-1 part the spec of gdf? or should we support both encodings?

cbrnr commented:
According to [this document](https://arxiv.org/pdf/cs/0608052v6,) the fields should be ASCII-encoded. Why doesn't it work with the default `decode` then (which uses UTF-8)?

agramfort commented:
no idea

cbrnr commented:
Hm, classic encoding problem. The test file I'm using happens to have an ö character in its header, that's why UTF-8 fails. We could default to UTF-8 and only in case of an error try latin-1, but really this could be any codepage...

larsoner commented:
I'm fine with utf8->latin1 fallback for now. We can deal with other codepages eventually if we need to by using `chardet` or something

agramfort commented:
fine with me

cbrnr commented:
But then again we can just use latin-1, because if utf-8 works (i.e. for a character in the first 7 bytes) then latin-1 produces exactly the same result. And we know for sure that the header fields are not encoded as utf-8.

larsoner commented:
Sure, if that works let's do it

cbrnr commented:
Yeah, it works, but now I remember why I didn't want to mess with file formats - there are so many instances where bytes are decoded, and currently we're assuming that they are ASCII. This might backfire with any new test file that gets into my hands... But OK, I guess this PR fixes the file I was currently working with, so it could be merged.

larsoner commented:
PEP8

https://travis-ci.org/mne-tools/mne-python/jobs/303521552#L2968

cbrnr commented:
Done, ready to merge.

agramfort commented:
you have a pep8 error line too long

cbrnr commented:
Ups, I've fixed this but this must have gotten lost during rebase. Done now (there were some comment also longer than 79 characters which I've also corrected just to be sure).

larsoner commented:
Thahnks @cbrnr
